examples/surrounding_nitrogen_bonds/random_selection.py

```diff
@@ -0,0 +1,61 @@
+"""Randomly selects N molecules from an input file.
+Usage:
+    python random_selection.py -n [N] --input [FILE] --output [FILE]
+"""
+
+import argparse
+import random
+
+from openeye import oechem
+
+
+def main():
+    """Reads in the file and does the selection."""
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        "-n",
+        type=int,
+        default=20,
+        help="Number of molecules to select.",
+    )
+    parser.add_argument(
+        "--input",
+        type=str,
+        help="Input molecule file.",
+    )
+    parser.add_argument(
+        "--output",
+        type=str,
+        help="Output molecule file.",
+    )
+    parser.add_argument(
+        "--seed",
+        type=int,
+        default=0,
+        help="Random seed.",
+    )
+    args = parser.parse_args()
+
+    subs = oechem.OESubSearch("[#7:1]-[#7:2]")
+    random.seed(args.seed)
+
+    # Input the molecules.
+    molecules = []
+    input_stream = oechem.oemolistream(args.input)
+    for mol in input_stream.GetOEMols():
+        if (subs.SingleMatch(mol)):
+            molecules.append(oechem.OEMol(mol))
+
+    #Choose the molecules.
+    #selected_molecules = random.sample(molecules, args.n)
+
+    #Output the molecules.
+    output_stream = oechem.oemolostream(args.output)
+    for mol in range(args.n):
+        randomNum = random.randint(0, len(molecules)-1)
```

Comment:
Doesn't this mean you can repeat molecules? It's quite likely you'll repeat molecules if you're selecting random indices each time (think of it as the birthday problem)

A better way would be to shuffle the molecules and choose the first n of them. For instance:

```python
molecules = random.shuffle(molecules)
for mol in molecules[:args.n]:
    ...
```